src/Entity/UrlButton.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\url_embed\Entity\UrlButton.
+ */
+
+namespace Drupal\url_embed\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\url_embed\UrlButtonInterface;
+
+/**
+ * Defines the UrlButton entity.
+ *
+ * @ConfigEntityType(
+ *   id = "url_button",
+ *   label = @Translation("Url Button"),
+ *   handlers = {
+ *     "list_builder" = "Drupal\url_embed\UrlButtonListBuilder",
+ *     "form" = {
+ *       "add" = "Drupal\url_embed\Form\UrlButtonForm",
+ *       "edit" = "Drupal\url_embed\Form\UrlButtonForm",
+ *       "delete" = "Drupal\Core\Entity\EntityDeleteForm"
+ *     }
+ *   },
+ *   config_prefix = "url_button",
+ *   admin_permission = "administer url buttons",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label",
+ *   },
+ *   config_export = {
+ *     "label",
+ *     "id",
+ *     "button_icon_uuid",
+ *   },
+ *   links = {
+ *     "edit-form" = "/admin/config/content/url-button/{url_button}",
+ *     "delete-form" = "/admin/config/content/url-button/{url_button}/delete"
+ *   }
+ * )
+ */
+class UrlButton extends ConfigEntityBase implements UrlButtonInterface {
+
+  /**
+   * The UrlButton ID.
+   *
+   * @var string
+   */
+  public $id;
+
+  /**
+   * Label of UrlButton.
+   *
+   * @var string
+   */
+  public $label;
+
+  /**
+   * UUID of the button's icon fili.
+   *
+   * @var string
+   */
+  public $button_icon_uuid;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getButtonImage() {
+    if ($this->button_icon_uuid && $image = $this->entityManager()->loadEntityByUuid('file', $this->button_icon_uuid)) {
+      return $image->url();
+    }
+    else {
+      return file_create_url(drupal_get_path('module', 'url_embed') . '/js/plugins/drupalurl/entity.png');
```

Comment:
Should we file this as a follow-up that we shouldn't be re-using the Entity Embed image but should be using something different?